### PR TITLE
Upgrade kubernetes-python client to 12.0.1

### PR DESCRIPTION
**Pull request by soneillf5**
### Proposed changes
Bump the version of the Kubernetes client from 8.0.1 to 12.0.1

The only change I could find was the removal of the delete options argument.

Created this PR to confirm the tests still pass.

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto master
- [x] I will ensure my PR is targeting the master branch and pulling from my branch from my own fork

### CODE REVIEW

**Review by ciarams87 — APPROVED** [review]
LGTM. Nice to say goodbye to those beta CRDs!!

File: tests/suite/custom_resources_utils.py [hunk 1]
```diff
@@ -214,10 +214,9 @@ def delete_policy(custom_objects: CustomObjectsApi, name, namespace) -> None:
     :return:
     """
     print(f"Delete a Policy: {name}")
-    delete_options = client.V1DeleteOptions()
 
     custom_objects.delete_namespaced_custom_object(
-        "k8s.nginx.org", "v1", namespace, "policies", name, delete_options
+        "k8s.nginx.org", "v1", namespace, "policies", name
```
Comment (vepatel) [inline tests/suite/custom_resources_utils.py:219]: are delete_options not needed unlike https://github.com/nginxinc/kubernetes-ingress/pull/1522/files#diff-b419599ec413dec04c511421059de8ee7b224c490b5785df0dcf2edb50394217R674-R679?

File: deployments/common/crds/appprotect.f5.com_aplogconfs.yaml [hunk 2]
```diff
@@ -15,7 +15,7 @@ spec:
   preserveUnknownFields: false
   scope: Namespaced
   versions:
-    - name: v1beta1
+    - name: v1
```
Comment (pleshakov) [inline deployments/common/crds/appprotect.f5.com_aplogconfs.yaml:18]: Hi @soneillf5 

Could you revert this field to v1beta1 in all AppProtect related resources? 
Note that this version is not related to the version of the CRD:
```
apiVersion: apiextensions.k8s.io/v1
kind: CustomResourceDefinition
...
```
In contrast, the `v1beta1` is the version of an AppProtect resource itself.
Once App Protect folks are happy to graduate the resources from `v1beta1` to `v1`, we can graduate them.

File: tests/settings.py [hunk 3]
```diff
@@ -8,7 +8,7 @@
 TEST_DATA = f"{PROJECT_ROOT}/data"
 DEFAULT_IMAGE = "nginx/nginx-ingress:edge"
 DEFAULT_PULL_POLICY = "IfNotPresent"
-DEFAULT_IC_TYPE = "nginx-ingress"
+DEFAULT_IC_TYPE = "nginx-plus-ingress"
```
[the remark this answers is not in the document]
Comment (soneillf5) [reply tests/settings.py:11]: fixed

File: deployments/common/crds/appprotect.f5.com_aplogconfs.yaml (hunk 2 above)
Comment (soneillf5) [reply deployments/common/crds/appprotect.f5.com_aplogconfs.yaml]: I've reverted the changes as you've suggested but can't get the tests to work now.

**Review by vepatel — APPROVED** [review]

File: tests/suite/test_app_protect_integration.py [hunk 4]
```diff
@@ -174,7 +174,7 @@ def test_ap_nginx_config_entries(
         for _ in conf_annotations:
             assert _ in result_conf
 
-    @pytest.mark.smoke
+    @pytest.mark.sean
```
Comment (pleshakov) [inline tests/suite/test_app_protect_integration.py:177]: looks like a temp change. was it supposed to be removed?

File: deployments/common/crds/appprotect.f5.com_aplogconfs.yaml (hunk 2 above)
Comment (vepatel) [reply deployments/common/crds/appprotect.f5.com_aplogconfs.yaml]: reverted and tests are passing now
